Index: b/src/http/ngx_http_upstream.c
===================================================================
--- a/src/http/ngx_http_upstream.c
+++ b/src/http/ngx_http_upstream.c
@@ -1592,6 +1592,11 @@ ngx_http_upstream_connect(ngx_http_reque
         return;
     }
 
+    if (rc >= NGX_HTTP_SPECIAL_RESPONSE) {
+        ngx_http_upstream_finalize_request(r, u, rc);
+        return;
+    }
+
     u->state->peer = u->peer.name;
 
 #if (NGX_HTTP_UPSTREAM_ZONE)
Index: b/src/http/ngx_http_upstream.h
===================================================================
--- a/src/http/ngx_http_upstream.h
+++ b/src/http/ngx_http_upstream.h
@@ -461,4 +461,9 @@ extern ngx_conf_bitmask_t  ngx_http_upst
 extern ngx_conf_bitmask_t  ngx_http_upstream_ignore_headers_masks[];
 
 
+#ifndef HAVE_BALANCER_STATUS_CODE_PATCH
+#define HAVE_BALANCER_STATUS_CODE_PATCH
+#endif
+
+
 #endif /* _NGX_HTTP_UPSTREAM_H_INCLUDED_ */
Index: b/src/stream/ngx_stream.h
===================================================================
--- a/src/stream/ngx_stream.h
+++ b/src/stream/ngx_stream.h
@@ -27,6 +27,7 @@ typedef struct ngx_stream_session_s  ngx
 
 
 #define NGX_STREAM_OK                        200
+#define NGX_STREAM_SPECIAL_RESPONSE          300
 #define NGX_STREAM_BAD_REQUEST               400
 #define NGX_STREAM_FORBIDDEN                 403
 #define NGX_STREAM_INTERNAL_SERVER_ERROR     500
Index: b/src/stream/ngx_stream_proxy_module.c
===================================================================
--- a/src/stream/ngx_stream_proxy_module.c
+++ b/src/stream/ngx_stream_proxy_module.c
@@ -769,6 +769,11 @@ ngx_stream_proxy_connect(ngx_stream_sess
         return;
     }
 
+    if (rc >= NGX_STREAM_SPECIAL_RESPONSE) {
+        ngx_stream_proxy_finalize(s, rc);
+        return;
+    }
+
     u->state->peer = u->peer.name;
 
 #if (NGX_STREAM_UPSTREAM_ZONE)
Index: b/src/stream/ngx_stream_upstream.h
===================================================================
--- a/src/stream/ngx_stream_upstream.h
+++ b/src/stream/ngx_stream_upstream.h
@@ -162,4 +162,9 @@ ngx_stream_upstream_srv_conf_t *ngx_stre
 extern ngx_module_t  ngx_stream_upstream_module;
 
 
+#ifndef HAVE_BALANCER_STATUS_CODE_PATCH
+#define HAVE_BALANCER_STATUS_CODE_PATCH
+#endif
+
+
 #endif /* _NGX_STREAM_UPSTREAM_H_INCLUDED_ */
